jena-arq/src/main/java/org/apache/jena/sparql/expr/nodevalue/XSDFuncOp.java

```diff
@@ -1554,4 +1551,55 @@ private static Duration valueCanonicalDuration(NodeValue nv) {
 //            dur = ... 
         return dur ;
     }
+
+    public static NodeValue adjustDatetimeToTimezone(NodeValue nv1,NodeValue nv2){
+        if(nv1 == null)
+            return null;
+
+        if(!nv1.isDateTime() && !nv1.isDate() && !nv1.isTime()){
+            throw new ExprEvalException("Not a valid date, datetime or time:"+nv1);
+        }
+
+        XMLGregorianCalendar calValue = nv1.getDateTime();
+        Boolean hasTz = calValue.getTimezone() != DatatypeConstants.FIELD_UNDEFINED;
+        int inputOffset = 0;
+        if(hasTz){
+            inputOffset = calValue.getTimezone();
+        }
+
+        int tzOffset = TimeZone.getDefault().getRawOffset() / (1000*60);
+        if(nv2 != null){
```

Comment:
`int tzOffset = TimeZone.getDefault().getOffset(new Date().getTime())/(1000*60) ;`

Ideally, don't set this with a call to `new Date()` until later -- see below.